packages/core/src/tech/HlsJsTech.ts

```diff
@@ -75,6 +79,34 @@ export default class HlsJsTech extends BaseTech {
     this.playlistDuration = data?.details?.totalduration;
   }
 
+  getVideoQualities() {
+    if (this.hls) {
+      return this.hls.levels.map((level) => {
+        const quality: IVideoQuality = {
+          width: level['width'],
+          height: level['height'],
+          bitrate: level['bitrate'],
+        };
+        return quality;
+      });
+    }
+  }
+
+  get currentLevel() {
+    if (this.hls) return this.hls.currentLevel;
```

Comment:
Please return the full level instead, basically the `IVideoQuality` item ( that should be renamed `IVideoLevel` )